text/0110-track-reform.md

```diff
@@ -21,15 +22,16 @@ Another confusion is when for `@wire` decorator reactivity works fine even when
 
 ## Proposal
 
-### "Unlearn" the `@track` decorator
+### Learn the `@track` decorator when you really need it.
 
-We don't have the concept of deprecate, instead it is more accurate to call it "unlearn" because new users don't need to learn about it, and old users can just forget about it. Components that are using that decorator will just simply continue to function, and the compiler will completely ignore (remove) that decorator.
+The runtime/compiler will figure what is the list of fields that are defined by a class, and make the template react to mutations of the field at runtime.
+
+Mutations on fields (ex: `this.x = 5;`) is the majority of the use cases, but sometimes you need a field to hold a value other than a primitive, and make changes in its value ex: `this.address.city = 'San Francisco'`. Only in such cases you will need to use the `@track` decorator.
 
-Instead, the runtime/compiler will figure what is the list of fields that are defined by a class, and make them all trackable at runtime.
 
 ### Compiler changes
 
-When the compiler compiles the class, it can extract any field that is not decorated with `@api`, and pass the metadata through the registerComponent call, this information will be used as the same list of things to be tracked today. Not need to change the logic in the engine, just the error messages if any.
+When the compiler compiles the class, it can extract any field that is not decorated with `@api`, `@track` or `@wire`, and pass the metadata through the registerComponent call. For every class property that is passed to the registerComponent, we will create a getter and a setter in the prototype in order to observe mutations in to class property. No change is needed to the logic in the engine.
```

Comment:
class property -> class field

let's not use property/properties that much, because that includes expandos, while fields are more clear.